### DISCUSSION TITLE
Thank you for your work... How about a push to Hubitat?

### DISCUSSION THREAD

**Discussion by TaZZaT** [jertel/vuegraf#31] [Ideas]
@jertel 

Thanks for the awesome work.  Was hoping I could work with you to add a push to Hubitat.  I have it working but wanted to check with you before I do a pull request.

Here's what I've done that works well with you code.

I added the following to vuegraf.json.

    "hubitat": {
        "host": "192.168.0.7",
        "port": 39501,
	"enable": true
    },

Then I added a a few lines of python code that sends hubitat the average usage every (INTERVAL_SEC).  It only runs this extra code (push to Hubitat) if config['hubitat']['enable'] = true

I've written the drivers (Parent and Child) that makes this work on the Hubitat Hub.

Thanks in advance,

Paul

**Comment by jertel** [comment]
Interesting idea. When I first read this I thought 'Why not pull the values out of the InfluxDB?', but that then puts the burden onto your Hubitat drivers to maintain InfluxDB credentials and knowledge about Influx query language as well as implement a timing loop to perform the periodic polling. So I can understand why it's simpler to request the Vuegraf 'push' it to Hubitat. If we go this route then it should be done in a modular way since others may want the same integration to their systems, such as pushing data to other smart hubs. Taking it further, InfluxDB could actually be it's own module as well. We wouldn't need to do this right now, but it helps show the path where this could go down the road.

**Reply by TaZZaT** [reply]
Thanks for the reply.  I originally built code to read from InfluxDB and push to Hubitat.  It worked but values were always way behind.  Then I looked at another package I'm using (tagyoureit/nodejs-poolController) and their philosophy is to just enable the pushes to different automation platforms by enabling/configuring the config.json file.  As for a module, I'm a bit of a hack when it comes to python programing (still learning) so I'll have to learn how to do that... haha.  Given that, I think it's a great idea...  let me move my code to module and I'll post it in a few days for your thoughts.

**Reply by jertel** [reply]
No problem. It doesn't need to be perfect. The main goal from my "modular" statement is to keep all of the Hubitat specific code inside of it's own hubitat.py file. Then in the main Vuegraf loop you will add a single line, something like `sendDataToHubitat(data)` function. It's good enough, and lends itself to easier refactoring should we ever want to fully modularize all destination endpoints.

**Comment by Robert-D** [comment]
+1 here for a Hubitat push, this would be awesome.
